azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -383,39 +391,94 @@ private OSProfile createOsProfile(String computerName, Template template) {
       return builder.build();
    }
 
-   private NetworkInterfaceCard createNetworkInterfaceCard(String subnetId, String name, String locationName,
-                                                           String azureGroup, TemplateOptions options) {
-      final PublicIPAddressApi ipApi = api.getPublicIPAddressApi(azureGroup);
+   private List<NetworkInterfaceCard> createNICs(String nodeName, String location, AzureTemplateOptions options) {
+      ImmutableList.Builder<NetworkInterfaceCard> nics = ImmutableList.builder();
+      String securityGroup = getOnlyElement(options.getGroups(), null);
 
-      PublicIPAddressProperties properties = PublicIPAddressProperties.builder().publicIPAllocationMethod("Static")
-              .idleTimeoutInMinutes(4).build();
+      for (IpOptions ipConfig : publicIpsFirst(options.getIpOptions())) {
```

Comment:
Not sure a function would be that appropriate here. This method (as a private method) depends a lot on the current context. It not only depends on the ip configurations, but also on other template options such as the security groups and the name of the node. I think it makes sense here as a private method.